src/App.js

```diff
@@ -1,14 +1,89 @@
-import React from 'react';
+import React from "react";
+import TodoForm from "./components/TodoComponents/TodoForm";
+import TodoList from "./components/TodoComponents/TodoList";
+import SimpleStorage from "react-simple-storage";
+
+import "./minireset.css";
+import "./App.css";
+
+const todos = [
+  {
+    task: "Organize Garage",
+    id: 1528817077286,
+    completed: false
+  },
+  {
+    task: "Bake Cookies",
+    id: 1528817084358,
+    completed: false
+  }
+];
 
 class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
+  constructor() {
+    super();
+    this.state = {
+      todos: todos,
+      todoInput: ""
+    };
+  }
+
+  handleChanges = e => {
+    this.setState({ [e.target.name]: e.target.value });
+  };
+
+  addTodo = e => {
+    e.preventDefault();
+
+    this.setState(state => ({
+      todos: [...state.todos, {
+        task: state.todoInput,
+        id: Date.now(),
+        completed: false
+      }],
+      todoInput: ""
+    }));
+  };
+
+  toggleComplete = (e, id) => {
+    this.setState(state => ({
+      todos: state.todos.map(todo =>
+        todo.id === id
+          ? {
+              ...todo,
+              completed: !todo.completed
+            }
+          : todo
+      )
+    }));
+  };
```

Comment:
I realise I don't need the `e` parameter here - I left it in by accident as I was using it in an earlier iteration. 😅